pkg/controller/vsphere/reconciler_test.go

```diff
@@ -1746,6 +1780,130 @@ func TestDelete(t *testing.T) {
+		devices.AssignController(additionalDisk, scsi.(types.BaseVirtualController))
+
+		err = vmObj.AddDevice(ctx, additionalDisk)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	extraDisksAttachedTestCases := []struct {
+		name        string
+		machine     func(t *testing.T, simServerHost string) *machinev1.Machine
+		node        func(t *testing.T) *corev1.Node
+		attachDisks bool
+		errMessage  string
+	}{
+		{
+			name:        "extra disk attached",
+			attachDisks: true,
+			machine: func(t *testing.T, simServerHost string) *machinev1.Machine {
+				return getMachineWithStatus(t, machinev1.MachineStatus{
+					NodeRef: &corev1.ObjectReference{
+						Name: nodeName,
+					},
+				}, simServerHost)
+			},
+			node: func(t *testing.T) *corev1.Node {
+				return getNodeWithConditions([]corev1.NodeCondition{
+					{
+						Type:   corev1.NodeReady,
+						Status: corev1.ConditionTrue,
+					},
+				})
+			},
+			errMessage: "additional attached disks detected, block vm destruction and wait for disks to be detached",
+		},
+		{
+			name:        "extra disk attached with no drain annotatiom",
+			attachDisks: true,
+			machine: func(t *testing.T, simServerHost string) *machinev1.Machine {
+				machine := getMachineWithStatus(t, machinev1.MachineStatus{
+					NodeRef: &corev1.ObjectReference{
+						Name: nodeName,
+					},
+				}, simServerHost)
+				machine.ObjectMeta.Annotations = map[string]string{
+					machinecontroller.ExcludeNodeDrainingAnnotation: "",
+				}
+				return machine
+			},
+			node: func(t *testing.T) *corev1.Node {
+				return getNodeWithConditions([]corev1.NodeCondition{
+					{
+						Type:   corev1.NodeReady,
+						Status: corev1.ConditionTrue,
+					},
+				})
+			},
+			errMessage: "disks were detached, vm will be attempted to destroy in next reconciliation, requeuing",
+		},
+	}
+	for _, tc := range extraDisksAttachedTestCases {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			_, sess, srv := initSimulator(t)
+			simParams, err := getVcenterSimParams(srv, namespace)
+			g.Expect(err).NotTo(HaveOccurred())
+
+			vm := simulator.Map.Any("VirtualMachine").(*simulator.VirtualMachine)
+			vm.Config.InstanceUuid = instanceUUID
+
+			if tc.attachDisks {
+				simClient := sess.Client.Client
+				g.Expect(addDiskToVm(context.TODO(), vm, fmt.Sprintf("%s-%s", vm.Name, tc.name), simClient)).To(Succeed())
+			}
+
+			client := fake.NewClientBuilder().WithScheme(scheme.Scheme).WithRuntimeObjects(
+				simParams.secret,
+				tc.machine(t, simParams.host),
+				simParams.configMap,
+				simParams.infra,
+				tc.node(t)).Build()
+			mScope, err := newMachineScope(machineScopeParams{
+				client:    client,
+				Context:   context.Background(),
+				machine:   tc.machine(t, simParams.host),
+				apiReader: client,
+			})
+			g.Expect(err).NotTo(HaveOccurred())
+
+			reconciler := newReconciler(mScope)
+
+			// expect the first call to delete to make the vSphere destroy request
```

Comment:
Is destroy correct or just power off?

Reply:
Copypaste artifact, the original comment above is not correct as well. Fixed it.